### PR TITLE
Remove some TravisCI configuration

### PR DESCRIPTION
**Pull request by exarkun**
Fixes: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3330
Fixes: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3331
Fixes: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3332
Fixes: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3333
Fixes: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3334
Fixes: https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3335

### CODE REVIEW

**Comment by exarkun** [comment]
The decreased coverage, as reported by codecov, might seem concerning given that this PR is about deleting CI jobs.  However, the specific changes look like the usual random noise to me.  I also don't see how those specific lines could be meaningfully changed by the removal of the CI this PR removes.

File: .travis.yml [hunk 1]
```diff
@@ -12,26 +12,14 @@ env:
...
     tox -e ${T}
-    # To verify that the resultant PyInstaller-generated binary executes
-    # cleanly (i.e., that it terminates with an exit code of 0 and isn't
-    # failing due to import/packaging-related errors, etc.).
-    if [ "${T}" = "pyinstaller" ]; then dist/Tahoe-LAFS/tahoe --version; fi
-
-after_success:
-  - if [ "${T}" = "coverage" ]; then codecov; fi
 
 notifications:
```
Comment (sajith) [inline .travis.yml:24]: Perhaps this too can go?  IRC notification seem to be some kind of vestigial business that currently doesn't serve a purpose.

Comment (sajith) [inline .travis.yml:24]: Perhaps this too can go?  IRC notification seem to be some kind of vestigial business that currently does not serve a purpose.

**Comment by sajith** [comment]
I tried to say something about IRC notifications from Travis, and GitHub chose to highlight something else.  Deleted that comment, and trying again.

Perhaps the `notifications` item also can go from Travis configuration?  IRC notification seem to be some kind of vestigial business that currently does not serve a purpose.

**Comment by exarkun** [comment]
> I tried to say something about IRC notifications from Travis, and GitHub chose to highlight something else. Deleted that comment, and trying again. About this bit:
> 
> https://github.com/tahoe-lafs/tahoe-lafs/blob/cfe70335683f0be269058da8c331516de65199be/.travis.yml#L24-L33
> 
> Perhaps this `notifications` item also can go away from Travis configuration? IRC notification seem to be some kind of vestigial business that currently does not serve a purpose.

Yea, that seems reasonable.  I have a follow-up PR in which I'm trying to move the Python 3 job to CircleCI also.  I think that PR is going to completely delete the travis config so that will take care of this.

**Pull request merged**